### PR TITLE
Adding postgres syntax support for filter arguments

### PR DESCRIPTION
**Pull request by rodriguise**
This PR defines an optional `Dialect` configuration setting and interface to provide "custom" formatting for filter parameters.

The provided implementation is for Postgres will generate arguments in the format of `col = $1, col2 = $2, colN = $n`.

This is useful when not using an orm like gorm, etc. Configuration might look like this:

```go
parser := rql.MustNewParser(rql.Config{
		Model:    Application{},
		ColumnFn: swag.ToJSONName,
		Dialect:  rql.Postgres(),
	})
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,9 @@
 module github.com/a8m/rql
 
-require github.com/mailru/easyjson v0.0.0-20180823135443-60711f1a8329
+require (
+	github.com/go-sql-driver/mysql v1.5.0
+	github.com/jinzhu/gorm v1.9.14
```
Comment (a8m) [inline go.mod:5]: Remove these 2 please.

File: rql.go [hunk 2]
```diff
@@ -349,6 +354,7 @@ type parseState struct {
 	*Parser                     // reference of the parser config
 	*bytes.Buffer               // query builder
 	values        []interface{} // query values
+	argN          int           // current arg counter
```
Comment (a8m) [inline rql.go:357]: Where do you increment it?

File: go.mod (hunk 1 above)
Comment (rodriguise) [reply go.mod:5]: `go mod tidy` picked those up. Will remove.

**Comment by rodriguise** [comment]
I thought about this and its overcomplicated...I am going to pull it back and resubmit shortly. Thanks for the review.

**Pull request closed without merging**